src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py

```diff
@@ -54,11 +54,40 @@ class NFApplicationConfiguration:
     name: str
     chartName: str
     chartVersion: str
+    releaseName: str
     dependsOnProfile: List[str]
     registryValuesPaths: List[str]
     imagePullSecretsValuesPaths: List[str]
     valueMappingsFile: str
 
+    def __post_init__(self):
+        """Format the fields based on the NFDV validation rules."""
+        self._format_name()
+        self._format_release_name()
+
+    def _format_name(self):
+        """
+        Format the name field.
+
+        The name should start with a alphabetic character, have alphanumeric characters
+        or '-' in-between and end with alphanumerc character, and be less than 64
+        characters long. See NfdVersionValidationHelper.cs in pez codebase
+        """
+        self.name = re.sub("[^0-9a-zA-Z-]+", "-", self.name)
```

Comment:
Could you add a comment here about what this is doing?  I think it is replacing any non-alphanumeric characters with a hyphen.